### PR TITLE
Add Input.Password

### PR DESCRIPTION
**Pull request by orzyyyy**
Ref: #13267

### CODE REVIEW

File: components/input/Password.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import Input, { InputProps } from './Input';
+import Icon from '../icon';
+
+export interface PasswordProps extends InputProps {
+  inputPrefixCls?: string;
```
Comment (ztplz) [inline components/input/Password.tsx]: Both of them are never reassigned, mark them with `readonly`.

**Review by ztplz — APPROVED** [review]

File: components/input/Password.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  action?: string;
+}
+
+export interface PreviewState {
```
Comment (ztplz) [inline components/input/Password.tsx]: Just some advise.
``````
export interface PasswordState {
  visible: boolean,
}

const ActionMap: Record<string, string> = {
  click: 'onClick',
  hover: 'onMouseOver',
};

export default class Password extends React.Component<PasswordProps, any> {
  static defaultProps = {
    inputPrefixCls: 'ant-input',
    prefixCls: 'ant-input-password',
    action: 'click',
  };
   state: PasswordState = {
     visible: false,
  };
   private input: Input;
   focus() {
    this.input.focus();
  }
   blur() {
    this.input.blur();
  }
   saveInput = (node: Input) => {
    this.input = node;
  }
   onChange = () => {
     this.input.blur();
     this.setState({
       visible: !this.state.visible,
     });
  }
   getIcon() {
    const { prefixCls, action } = this.props;
     const iconTrigger = ActionMap[action] || '';
    const iconProps = Object.assign(
      {},
      {
        className: `${prefixCls}-icon`,
        type: "eye",
        theme: this.state.visible ? 'filled' : undefined,
        key: "passwordIcon",
        [iconTrigger]: this.onChange,
      },
    )
     return React.cloneElement(<Icon {...iconProps} />);
  }
   render() {
    const { className, prefixCls, inputPrefixCls, size, suffix, ...restProps } = this.props;
    const suffixIcon = this.getIcon();
    const inputClassName = classNames(prefixCls, className, {
      [`${prefixCls}-${size}`]: !!size,
    });
     return (
      <Input
        {...restProps}
        type={this.state.visible ? 'text' : 'password'}
        size={size}
        className={inputClassName}
        prefixCls={inputPrefixCls}
        suffix={suffixIcon}
        ref={this.saveInput}
      />
    );
  }
}
``````

**Review by ztplz — APPROVED** [review]

File: components/input/Password.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  theme: any,
+  type: string,
+}
+
+export default class Password extends React.Component<PasswordProps, any> {
```
Comment (ztplz) [inline components/input/Password.tsx]: `export default class Password extends React.Component<PasswordProps, PasswordState> {`

**Review by ztplz — APPROVED** [review]

File: components/input/Password.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  static defaultProps = {
+    inputPrefixCls: 'ant-input',
+    prefixCls: 'ant-input-password',
+    action: 'click',
+  };
+
+  state: PreviewState = {
+    theme: undefined,
+    type: "password",
```
Comment (ilanus) [inline components/input/Password.tsx]:
```diff
- type: "password",
+ type: 'password',
```

File: components/input/Password.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  onChange = () => {
+    const { theme } = this.state;
+    let inputState = {};
+
+    this.input.blur();
+
+    if(theme === 'filled') {
+      inputState = {
+        theme: undefined,
+        type: "password",
```
Comment (ilanus) [inline components/input/Password.tsx]:
```diff
- type: "password",
+ type: 'password',
```

File: components/input/Password.tsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+      };
+    } else {
+      inputState = {
+        theme: "filled",
```
Comment (ilanus) [inline components/input/Password.tsx]:
```diff
- type: "filled",
+ type: 'filled',
```

File: components/input/Password.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        type: "text",
```
Comment (ilanus) [inline components/input/Password.tsx]:
```diff
- type: "text",
+ type: 'text',
```

File: components/input/Password.tsx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+    this.input = node;
+  }
+
+  onChange = () => {
+    const { theme } = this.state;
+    let inputState = {};
+
+    this.input.blur();
+
+    if(theme === 'filled') {
```
Comment (ilanus) [inline components/input/Password.tsx]:
```diff
- if(theme === 'filled')
+ if (theme === 'filled')
```

File: components/input/Password.tsx [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+  getIcon() {
+    const { prefixCls } = this.props;
+
+    const iconTrigger = this.getTrigger();
+    const iconProps = Object.assign(
+      {},
+      {
+        className: `${prefixCls}-icon`,
+        type: "eye",
```
Comment (ilanus) [inline components/input/Password.tsx]:
```diff
- type: "eye",
+ type: 'eye',
```

File: components/input/Password.tsx [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+        theme: this.state.theme,
+        key: "passwordIcon",
```
Comment (ilanus) [inline components/input/Password.tsx]:
```diff
- type: "passwordIcon",
+ type: 'passwordIcon',
```

File: components/input/Password.tsx [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+        [iconTrigger]: this.onChange,
+      },
+    )
```
Comment (ilanus) [inline components/input/Password.tsx]:
```
;
```

**Comment by zy410419243** [comment]
...I'll create a new `PR`...I always forget update the branch which I rebase from up to date...

**Pull request closed without merging**